reproject/interpolation/core_celestial.py

```diff
@@ -23,7 +25,8 @@ def _reproject_celestial(array, wcs_in, wcs_out, shape_out, order=1):
     """
 
     # Make sure image is floating point
-    array = np.asarray(array, dtype=float)
+    if not np.issubdtype(array.dtype, np.float):
+        array = np.asarray(array, dtype=float)
```

Comment:
@keflavich The docs state that:

> No copy is performed if the input is already an ndarray. If a is a subclass of ndarray, a base class ndarray is returned.

(https://docs.scipy.org/doc/numpy/reference/generated/numpy.asarray.html#numpy-asarray)

But yeah, the endianess can be a problem, especially because FITS always returns the "wrong" one (at least on my computer 😄 )